src/RA_Interface.h

```diff
@@ -2,102 +2,139 @@
 #define RA_INTERFACE_H
 #pragma once
 
-//	NB. Shared between RA_Integration and emulator
-using BOOL = int;
+/* NB. Shared between RA_Integration and emulator */
+
+/*
+    NB. If using MSVC, the compiler switch '/Zc:__cplusplus' must be added manually to correctly identifiy the 
+    C++ standard supported. It doesn't fully implement C99 so __STDC_VERSION__ won't work here but should work
+    with other compilers (GCC/Clang/etc.).
+
+    Visual Studio only has values for C++14/17, but these are standard macros supported by most compilers
+*/
+#if (!defined(__cplusplus) && (defined(__STDC__) && !defined(__STDC_HOSTED__) && !defined(__STDC_VERSION__)))
+
+/* vvv C89 only vvv */
+#define bool  int
+#define false 0
+#define true  1 
+#define inline /* nothing, inline didn't exist until C99 */
+
+/* ^^^ C89 only ^^^ // vvv C89 or C99 vvv  */
+#elif (!defined(__cplusplus) && (((defined(__STDC_HOSTED__) || (__STDC_VERSION__  >= 199901L))) || defined(__STDC__)))
+#define _ENUM_CLASS typedef enum
+#define noexcept /* ^^^ C89 or C99 ^^^ // vvv C99 only vvv */
+
+#elif (!defined(__cplusplus) && !defined(__STDC__)) && (__STDC_HOSTED__ || (__STDC_VERSION__ >= 199901L))
+#include <stdbool.h>
+
+#elif (__cplusplus < 201103L) /* ^^^ C99 only ^^^ // vvv pre-C++11 vvv */
+#define _ENUM_CLASS enum
+#define noexcept throw() /* error prone version of noexcept, used a type list instead of nothrow condition */
+
+#else /* ^^^ pre-C++11 ^^^  // vvv C++11 onwards vvv */
+#define _ENUM_CLASS enum class
+#endif /* (!defined(__cplusplus) && (defined(__STDC__) && !defined(__STDC_HOSTED__) && !defined(__STDC_VERSION__))) */
 
 #ifndef CCONV
 #define CCONV __cdecl
 #endif
 
 struct ControllerInput
 {
-    BOOL m_bUpPressed;
-    BOOL m_bDownPressed;
-    BOOL m_bLeftPressed;
-    BOOL m_bRightPressed;
-    BOOL m_bConfirmPressed;	//	Usually C or A
-    BOOL m_bCancelPressed;	//	Usually B
-    BOOL m_bQuitPressed;	//	Usually Start
+#if (defined(__cplusplus) || (defined(__STDC__) || defined(__STDC_VERSION__)))
+#if ((__cplusplus < 201103L) || (__STDC__ || (__STDC_VERSION__ >= 199901L)))
```

Comment:
instead of repeating this logic, could you put something at the top of the file that defines capabilities and use those?
```
#if (__STDC_VERSION__ >= 199901L)
  #define HAS_C99 1
  #if (defined(__cplusplus) && (__cplusplus >= 201103L))
    #define HAS_CPP11 1
  #endif
#endif
```
This logic may not be perfect, but I think you get the idea.

Then you could make the processing logic easier to read:
```
#if HAS_CPP11
  #define _ENUM_CLASS enum class
#elif HAS_C99
  #define _ENUM_CLASS enum
#else
  #define _ENUM_CLASS typedef enum
#endif
```

Reply:
alright, I'm going to try doing your suggestions. Gonna close this PR for now.